src/components/Accordion/Accordion.tsx

```diff
@@ -18,50 +19,74 @@ const pickOpen: (items: AccordionItem[]) => AccordionItemId[] = pipe(
   pluck('id'),
 );
 
-const Accordion: React.FC<AccordionProps> = ({
-  isCollapsedOnOpen = true,
-  items,
-  isCard = true,
-  className,
-  ...props
-}) => {
-  const [openIds, setOpenIds] = useState(pickOpen(items));
+function filterState(
+  state: AccordionItemId[],
+  item: AccordionItemId,
+  isCollapsedOnOpen: boolean,
+) {
+  if (includes(item, state)) {
+    return reject(equals(item), state);
+  }
+  if (isCollapsedOnOpen) {
+    return [item];
+  }
+  return [...state, item];
+}
 
-  const handleClick = useCallback(
-    (id: AccordionItemId) => {
-      setOpenIds((state) =>
-        includes(id, state)
-          ? reject(equals(id), state)
-          : isCollapsedOnOpen
-          ? [id]
-          : [...state, id],
-      );
+const Accordion = React.forwardRef<HTMLDivElement, AccordionProps>(
+  (
+    {
+      isCollapsedOnOpen = true,
+      items,
+      openItems,
+      isCard = true,
+      className,
+      ...props
     },
-    [setOpenIds, isCollapsedOnOpen],
-  );
+    ref,
+  ) => {
+    const [openIds, setOpenIds] = useState(pickOpen(items));
 
-  return (
-    <Stack {...props} className={cls(CLX_COMPONENT, className)}>
-      {items.map(({ title, id, content }) => (
-        <AccordionCollapsible
-          key={`accordion-item-${id}`}
-          handleHeaderClick={handleClick}
-          id={id}
-          isCard={isCard}
-          isOpen={includes(id, openIds)}
-          title={title}
-        >
-          {content}
-        </AccordionCollapsible>
-      ))}
-    </Stack>
-  );
-};
+    useEffect(
```

Comment:
`useDeepCompareEffect` might be better in this case since the `openItems` is an array and `useEffect` is checking reference equality so even if items are the same in the current and previous state the effect will be triggered in every rerender